builder/nutanix/driver.go

```diff
@@ -656,6 +656,44 @@ func (d *NutanixDriver) SaveVMDisk(diskUUID string, imageName string) (*nutanixI
 		return nil, fmt.Errorf("error while NewV3Client, %s", err.Error())
 	}
 
+	// When force_deregister, check if image already exists
+	if ForceDeregister {
+		log.Println("force_deregister is set, check if image already exists")
+		ImageList, err:=conn.V3.ListAllImage(fmt.Sprintf("name==%s",imageName))
+		if err != nil {
+			return nil, fmt.Errorf("error while ListAllImage, %s", err.Error())
+		}
+		if *ImageList.Metadata.TotalMatches==0 {
+			log.Println("Image with given Name not found, no need to deregister")
+		} else if *ImageList.Metadata.TotalMatches>1 {
+			log.Println("More than one image with given Name found, will not deregister")
+		} else if *ImageList.Metadata.TotalMatches==1 {
+			log.Println("Exactly one image with given Name found, will deregister")
+			resp,err:= conn.V3.DeleteImage(*ImageList.Entities[0].Metadata.UUID)
+			if err != nil {
+				return nil, fmt.Errorf("error while DeleteImage, %s", err.Error())
+			}
+			taskUUID := resp.Status.ExecutionContext.TaskUUID.(string)
+			log.Printf("Wait until delete Image %s is finished, %s\n",*ImageList.Entities[0].Metadata.UUID,taskUUID)
+			// Wait for the Image to be deleted
+			for {
```

Comment:
can you implement a timeout on the loop, here we can fall in an endless situation

Reply:
fixed